packages/core/src/external/quest/file/file-generator.ts

```diff
@@ -0,0 +1,100 @@
+import { z } from "zod";
+import { Patient } from "@metriport/shared/domain/patient";
+import { MetriportError } from "@metriport/shared";
+import { makeNameDemographics } from "@metriport/shared/common/demographics";
+import { makeGenderDemographics } from "../codes";
+
+import { RelationshipToSubscriber } from "../codes";
+import {
+  requestHeaderRow,
+  requestHeaderSchema,
+  requestDetailSchema,
+  requestDetailRow,
+  requestFooterSchema,
+  requestFooterRow,
+} from "../schema/request";
+import { OutgoingFileRowSchema } from "../schema/shared";
+
+export function buildRosterFile(patients: Patient[]): Buffer {
+  const requestedPatientIds: string[] = [];
+
+  const header = buildQuestRequestRow(
+    { recordType: "H", generalMnemonic: "METRIP", fileCreationDate: new Date() },
+    requestHeaderSchema,
+    requestHeaderRow
+  );
+
+  const details = patients.flatMap(patient => {
+    // Generate the request row for this patient
+    const row = buildPatientRow(patient);
+    if (row) {
+      requestedPatientIds.push(patient.id);
+      return [row];
+    }
+    return [];
+  });
```

Comment:
May be worth doing something like this to avoid mutating state from within a map/flatMap (a nono / unexpected thing)

```
const [details, patientIds] = _(patients)
  .map(patient => ({ patient, row: buildPatientRow(patient) }))
  .filter('row')
  .thru(validPatients => [
    _.map(validPatients, 'row'),
    _.map(validPatients, 'patient.id')
  ])
  .value();
```

Reply:
This is awesome!! Changing this now, and will keep my eyes open for other opportunities to leverage underscore 🙌